classes/items/itembase.cpp

```diff
@@ -18,44 +18,62 @@ void Item::generate()
 
 
 //Constructors / Destructors
-Item::Item()
-{
+Item::Item(std::string name,
+            unsigned int type,
+            unsigned int tier,
+            unsigned int value,
+            unsigned int level);
+
     std::string name;
-    unsigned type;
-    unsigned value;
+    unsigned int type;
+    unsigned int value;
+    unsigned int tier;
+    unsigned int level;
+
+    this->name = name; //assigns the given value of name
+    this->type = type;  //assigns the given value of type
+    this->value = value; //assigns the given value of value
+    this->tier = tier; //assigns the given value of tier.
+    this->level = level;
 
-    this->name = ""; //assigns the initial value of name to null
-    this->type = 0;  //assigns the initial value of type to 0
-    this->value = 0; //assigns the initial value of value to 0
-}
 
 Item::~Item()
 {
 
 }
 
 //Accessors
-const std::string& getName()
+const std::string& get_name()
 {
     //return this->name; //assigns a generated item its name
 }
-const unsigned& getType()
+const unsigned int& get_type()
```

Comment:
Same with the other ones. make sure to make it part of the Item:: scope and you no longer need the & as the this-> pointer is included via the scoping.